src/electronics/coulombs_law.rs

```diff
@@ -0,0 +1,183 @@
+///   Apply Coulomb's Law on any three given values. These can be force, charge1,
+///   charge2, or distance, and then in a Python dict return name/value pair of
+///   the zero value.
+///
+///   Coulomb's Law states that the magnitude of the electrostatic force of
+///   attraction or repulsion between two point charges is directly proportional
+///   to the product of the magnitudes of charges and inversely proportional to
+///   the square of the distance between them.
+///
+/// -> Wikipedia reference: https://en.wikipedia.org/wiki/Coulomb%27s_law
+
+const COULOMBS_CONSTANT: f64 = 8.9875517923e9;
+#[derive(PartialEq, Eq, Debug)]
+pub enum CoulombsLawError {
+    ExtraZeroArg(String),
+    NegativeDistance(String),
+}
+pub fn coulombs_law(
+    force: f64,
+    charge1: f64,
+    charge2: f64,
+    distance: f64,
+) -> Result<String, CoulombsLawError> {
+    let charge_product = (charge1 * charge2).abs();
+
+    if invalid_arguments(force, charge1, charge2, distance) {
+        return Err(CoulombsLawError::ExtraZeroArg(String::from(
+            "One and only one argument must be 0",
+        )));
+    }
+
+    if distance < 0.0 {
+        return Err(CoulombsLawError::NegativeDistance(String::from(
+            "Distance cannot be negative",
+        )));
+    }
+
+    if force == 0.0 {
+        return Ok(format!(
+            "force: {}",
+            calculate_force(charge_product, distance)
+        ));
+    } else if charge1 == 0.0 {
+        return Ok(format!(
+            "charge1: {}",
+            calculate_charge(charge2, force, distance)
+        ));
+    } else if charge2 == 0.0 {
+        return Ok(format!(
+            "charge2: {}",
+            calculate_charge(charge1, force, distance)
+        ));
+    }
+    Ok(format!(
+        "distance: {}",
+        calculate_distance(charge_product, force)
+    ))
+}
```

Comment:
Is this function really needed? If you insist on having it, please consider using `Optional` as inputs.

Reply:
based on electronic modules in python repository : https://github.com/TheAlgorithms/Python/blob/master/electronics/coulombs_law.py

Reply:
using optional would not cause boiler plate ?
in this situation i should double check value of inputs (none or some and sign of inputs)